packages/components/src/HeaderBar/HeaderBar.component.js

```diff
@@ -36,13 +36,25 @@ function Logo({ isFull, getComponent, t, ...props }) {
 	);
 }
 
-function Brand({ label, isSeparated, getComponent, ...props }) {
+function Brand({ label, isSeparated, getComponent, t, ...props }) {
 	const className = classNames(theme['tc-header-bar-action'], {
 		[theme.separated]: isSeparated,
 	});
 	const Renderers = Inject.getAll(getComponent, { ActionDropdown, Action });
-	const ActionComponent = props && props.items ? Renderers.ActionDropdown : Renderers.Action;
-	const clickAction = props && props.items ? undefined : props.onClick;
+
+	let ActionComponent;
+	let clickAction;
+	let ariaLabel;
```

Comment:
Just a question : Can we make something more functionnal oriented (immutable) with outside function to resolve your let variables ?

Reply:
I'm not a fan of creating a function that simple just to use a const.
Using a function will not change the complexity or readablity of the current one.